gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/RefreshTokenAuth2Provider.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2022 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.hadoop.fs.gcs;
```

Comment:
Probably better to move this class in the `auth` package and `hadoop-util` package, for consistency with rest Hadoop credential/auth classes?

Reply:
I initially did start in hadoop-util and hit one issue:
- I needed `GoogleCloudStorageOptions` for the proxy properties
Then also did move it in `package` but that required a change of method visibility for `GoogleCloudStorageOptions#getGcsOptionsBuilder`

Moving to `hadoop-util` will involve a bit of refactoring so perhaps I keep it where it is but in `auth` package (with a change of the method visibility)

Reply:
Note: With the implementation of your other comments around config, no need to change the `GoogleCloudStorageOptions#getGcsOptionsBuilder` method visibility at the end.

Reply:
Only things that block me to move it to the `hadoop-util` package now is `import static com.google.cloud.hadoop.fs.gcs.GoogleHadoopFileSystemConfiguration.CONFIG_KEY_PREFIXES;
`
If you got a suggestion on how to workaround this.

Reply:
This makes me think, that we may want to implement this auth method natively instead of relying on auth provider that was implemented for 3rd party integrations, especially if we want to use "native" GCS connector config options.

Reply:
true, that would be even better. It could simply be the default OAuth2 provider at the end of the day.

Reply:
Let me know how you want to proceed with that (ie where you want me to implement the refresh token logic)